eva-lib/src/main/java/uk/ac/ebi/variation/eva/lib/datastore/DBAdaptorConnector.java

```diff
@@ -64,7 +69,36 @@ public static VariantSourceDBAdaptor getVariantSourceDBAdaptor(String species)
         return new VariantSourceMongoDBAdaptor(getCredentials(species, properties),
                 properties.getProperty("eva.mongo.collections.files"));
     }
-    
+
+    public static MongoClient getMongoClient(Properties properties) throws UnknownHostException {
+
+        String authenticationDb = properties.getProperty("eva.mongo.auth.db", null);
+        if (authenticationDb == null || authenticationDb.isEmpty()) {
+            return new MongoClient();
+        } else {
+            String[] hosts = properties.getProperty("eva.mongo.host").split(",");
+            List<ServerAddress> servers = new ArrayList<>();
+
+            // Get the list of hosts (optionally including the port number)
+            for (String host : hosts) {
+                String[] params = host.split(":");
+                if (params.length > 1) {
+                    servers.add(new ServerAddress(params[0], Integer.parseInt(params[1])));
+                } else {
+                    servers.add(new ServerAddress(params[0], 27017));
+                }
+            }
```

Comment:
The hosts list generation could be extracted into a method because it's duplicated in `getMongoClient` and `getCredentials`.